packages/runtimes/hmr/src/loaders/hmr-runtime.js

```diff
@@ -40,7 +40,7 @@ var parent = module.bundle.parent;
 if ((!parent || !parent.isParcelRequire) && typeof WebSocket !== 'undefined') {
   var hostname = getHostname();
   var port = getPort();
-  var protocol = location.protocol === 'https:' ? 'wss' : 'ws';
+  var protocol = HMR_SECURE ? 'wss' : 'ws';
```

Comment:
This is sketchy if a user is, for example, running through ngrok. I couldn't think of any other way to make content scripts HMR work. Perhaps the behavior can be rewritten to choose HTTPS if `(HMR_SECURE || (location.protocol == 'https:' && hostname != 'localhost'))`.